db/change_log.go

```diff
@@ -253,34 +253,54 @@ func (conn *SqliteStreamDB) initTriggers(tableName string) error {
 	return nil
 }
 
+func (conn *SqliteStreamDB) filterChangesTo(changed chan fsnotify.Event, watcher *fsnotify.Watcher) {
+	for {
+		select {
+		case ev, ok := <-watcher.Events:
+			if !ok {
+				close(changed)
+				return
+			}
+
+			if ev.Op == fsnotify.Chmod {
+				continue
+			}
+
+			changed <- ev
+		}
+	}
+}
+
 func (conn *SqliteStreamDB) watchChanges(watcher *fsnotify.Watcher, path string) {
 	shmPath := path + "-shm"
 	walPath := path + "-wal"
 
 	errDB := watcher.Add(path)
 	errShm := watcher.Add(shmPath)
 	errWal := watcher.Add(walPath)
+	dbChanged := make(chan fsnotify.Event)
 
 	tickerDur := time.Duration(cfg.Config.PollingInterval) * time.Millisecond
 	changeLogTicker := utils.NewTimeoutPublisher(tickerDur)
 
 	// Publish change logs for any residual change logs before starting watcher
 	conn.publishChangeLog()
+	go conn.filterChangesTo(dbChanged, watcher)
 
 	for {
 		changeLogTicker.Reset()
 
 		err := conn.WithReadTx(func(_tx *sql.Tx) error {
 			select {
-			case ev, ok := <-watcher.Events:
+			case ev, ok := <-dbChanged:
```

Comment:
I don't see the point of adding another channel and doing the same thing that we are doing in the body of the function. This one new channel + go routine might not be a big deal, but unless there is sophisticated logic adding such additional go routines and channels is just gonna keep on increasing memory overhead.